unity-renderer/Assets/Scripts/MainScripts/DCL/Environment/Environment.cs

```diff
@@ -114,6 +115,15 @@ public void Dispose()
                 platform?.Dispose();
                 hud?.Dispose();
             }
+            public void OnDestroy()
```

Comment:
have you tried using `Dispose` method for flagging `IsBeingDestroyed` ?

Reply:
I needed a way to flag this before the `Dispose` as the PluginSystem is the first one to be disposed and needs to know if everything is being destroyed